Rubberduck.Refactorings/EncapsulateField/FieldCandidates/EncapsulateFieldCandidate.cs

```diff
@@ -21,217 +18,122 @@ public interface IEncapsulateFieldCandidate : IEncapsulateFieldRefactoringElemen
-        IValidateVBAIdentifiers NameValidator { set; get; }
         IEncapsulateFieldConflictFinder ConflictFinder { set; get; }
         bool TryValidateEncapsulationAttributes(out string errorMessage);
-        string IdentifierForReference(IdentifierReference idRef);
-        IEnumerable<PropertyAttributeSet> PropertyAttributeSets { get; }
     }
 
     public class EncapsulateFieldCandidate : IEncapsulateFieldCandidate
     {
-        protected Declaration _target;
-        protected QualifiedModuleName _qmn;
-        protected readonly string _uniqueID;
-        protected int _hashCode;
-        private string _identifierName;
+        protected readonly int _hashCode;
         protected EncapsulationIdentifiers _fieldAndProperty;
-        private string _rhsParameterIdentifierName;
 
-        public EncapsulateFieldCandidate(Declaration declaration, IValidateVBAIdentifiers identifierValidator)
+        public EncapsulateFieldCandidate(Declaration declaration)
         {
-            _target = declaration;
-            NameValidator = identifierValidator;
-            _rhsParameterIdentifierName = Resources.Refactorings.Refactorings.CodeBuilder_DefaultPropertyRHSParam;
+            Declaration = declaration;
+            AsTypeName = declaration.AsTypeName;
+
+            _fieldAndProperty = new EncapsulationIdentifiers(declaration.IdentifierName);
+            BackingIdentifierMutator = (value) => _fieldAndProperty.Field = value;
 
-            _fieldAndProperty = new EncapsulationIdentifiers(declaration.IdentifierName, identifierValidator);
             IdentifierName = declaration.IdentifierName;
-            PropertyAsTypeName = declaration.AsTypeName;
-            _qmn = declaration.QualifiedModuleName;
 
-            CanBeReadWrite = true;
+            TargetID = IdentifierName;
 
-            _uniqueID = $"{_qmn.Name}.{declaration.IdentifierName}";
-            _hashCode = _uniqueID.GetHashCode();
+            QualifiedModuleName = declaration.QualifiedModuleName;
 
-            ImplementLet = true;
-            ImplementSet = false;
-            if (_target.IsEnumField() && _target.AsTypeDeclaration.HasPrivateAccessibility())
-            {
-                //5.3.1 The declared type of a function declaration 
-                //may not be a private enum name.
-                PropertyAsTypeName = Tokens.Long;
-            }
-            else if (_target.AsTypeName.Equals(Tokens.Variant)
-                && !_target.IsArray)
-            {
-                ImplementSet = true;
-            }
-            else if (Declaration.IsObject)
-            {
-                ImplementLet = false;
-                ImplementSet = true;
-            }
-        }
+            //5.3.1 The declared type of a function declaration may not be a private enum.
+            PropertyAsTypeName = declaration.IsEnumField() && declaration.AsTypeDeclaration.HasPrivateAccessibility()
+                ? Tokens.Long
+                : declaration.AsTypeName;
```

Comment:
Will the preview issue a warning in this case? I'm inclined to think that it's usually a bug or at least unintentional to expose a member that corresponds to a private enum.

Reply:
In the current version, if the user selected a field for encapsulation (`Private` or `Public`), then the refactoring 'assumed' that the intent was to make them all publicly accessible.  A regrettable design choice in retrospect.  Exposing a `Private` field as a `Public Property` would be better implemented as an option on a per field basis.  #5421, which identifies this issue, was created based on your feedback (in chat IIRC) some time ago.  Since the focus this PR is on technical debt and changing this likely requires a change in the UI, I'd like to defer resolution of this comment to issue #5421.